app/controllers/follows_controller.rb

```diff
@@ -79,16 +79,25 @@ def create
     render json: { outcome: @result }
   end
 
-  def update
-    @follow = Follow.find(params[:id])
-    authorize @follow
-    redirect_to "/dashboard/following" if @follow.update(follow_params)
+  def bulk_update
+    @follows = Follow.where(id: params_for_update.keys).includes(:follower, :followable)
+    authorize @follows
+    Follow.transaction do
```

Comment:
should we write a test for this? Maybe by submitting a request with non decimal points? Something to make this explode

Reply:
It is a great idea to add a test here, but I cannot figure out a request to make it explode.
ActiveRecord will change the type of data when saving (here points is float so is `to_f`), so nondecimal points or even negative integers will pass. Besides, the array or hash will not pass the policy before the transaction, and I didn't' find any validation for the weight point.

Do you have any idea? 🙏

Reply:
That was the only one I had in mind. I don't think it's worth it to go down a rabbit hole of deep mocks. Can `follow.id` be invalid ?

Reply:
I cannot find either. I have tried none exist id or updating the same id twice, but none works. None exist id will be ignored when `Follow.where`, duplicated id will only cut into one when `params_for_update`.

Reply:
Ok, let's leave it like this in the meantime, I don't think it's super critical to have a test for the transaction that we know it's going to be rolled back. If you could address the other notes we can merge this afterwards. Thanks for bouncing ideas!